geochem_inverse_optimize.py

```diff
@@ -202,6 +208,12 @@ def solve(
 
         return downstream_preds, upstream_preds
 
+    def get_misfit(self) -> float:
+        return cp.norm(cp.vstack(self._primary_terms)).value
+
+    def get_roughness(self) -> float:
```

Comment:
I'm not sure if `roughness` is the best name for what this is returning (or `misfit` for the above). I'll approve anyway.